modules/data.atmosphere/R/met.process.R

```diff
@@ -259,20 +259,24 @@ met.process <- function(site, input_met, start_date, end_date, model, host, dbpa
                               slat=new.site$lat,slon=new.site$lon,newsite=new.site$id)
 
   }else if(register$scale=="site"){ ##### Site Level Processing
-
-    logger.info("Gapfilling") # Does NOT take place on browndog!
-
-    input.id   <- cf.id[1]
-    outfolder  <- file.path(dir,paste0(met,"_CF_gapfill_site_",str_ns))
-    pkg        <- "PEcAn.data.atmosphere"
-    fcn        <- "metgapfill"
-    formatname <- 'CF Meteorology'
-    mimetype   <- 'application/x-netcdf'
-    lst        <- site.lst(site,con)
-
-    ready.id   <- convert.input(input.id,outfolder,formatname,mimetype,site.id=site$id,start_date,end_date,pkg,fcn,
-                                username,con=con,hostname=host$name,browndog=NULL,write=TRUE,lst=lst)
-
+    if(!is.null(register$gapfill)){
```

Comment:
The problem with that approach is that metgapfill breaks on other met
products right now. It's too specific to ameriflux. In theory, any met
product that doesn't need to be gapfilled should pass right through and
this is most likely a fairly simple thing to generalize but I can't go off
on more tangents. I'm probably going to leave this quick fix until I can
revisit it.

On Thu, Nov 5, 2015, 10:14 PM Michael Dietze <notifications@github.com>
wrote:

> In modules/data.atmosphere/R/met.process.R
> <https://github.com/PecanProject/pecan/pull/671#discussion_r44099772>:
>
> > @@ -259,20 +259,24 @@ met.process <- function(site, input_met, start_date, end_date, model, host, dbpa
> >                                slat=new.site$lat,slon=new.site$lon,newsite=new.site$id)
> >
> >    }else if(register$scale=="site"){ ##### Site Level Processing
> > -
> > -    logger.info("Gapfilling") # Does NOT take place on browndog!
> > -
> > -    input.id   <- cf.id[1]
> > -    outfolder  <- file.path(dir,paste0(met,"_CF_gapfill_site_",str_ns))
> > -    pkg        <- "PEcAn.data.atmosphere"
> > -    fcn        <- "metgapfill"
> > -    formatname <- 'CF Meteorology'
> > -    mimetype   <- 'application/x-netcdf'
> > -    lst        <- site.lst(site,con)
> > -
> > -    ready.id   <- convert.input(input.id,outfolder,formatname,mimetype,site.id=site$id,start_date,end_date,pkg,fcn,
> > -                                username,con=con,hostname=host$name,browndog=NULL,write=TRUE,lst=lst)
> > -
> > +    if(!is.null(register$gapfill)){
>
> Yes, eventually in pecan.xml but for now it's probably sufficient to put
> it as an arguement to met.process and set the default to the current
> function since at the moment we don't have any alternative gapfilling
> schemes implemented
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/PecanProject/pecan/pull/671/files#r44099772>.
>